.github/workflows/auto-label.yml

```diff
@@ -0,0 +1,35 @@
+name: Label issues
+on:
+  issues:
+    types:
+      - opened
+
+jobs:
+  label_issues:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    env:
+      ISSUE_BODY: ${{ toJson(github.event.issue.body) }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Check for program keywords in description
+        id: check-programs
+        run: |
+          PROGRAMS=$(jq -r '.programs | join(",")' programs.json)
+          IFS=',' read -r -a programs <<< "$PROGRAMS"
+          for program in "${programs[@]}"; do
+            if echo "$ISSUE_BODY" | grep -q -i "${program}"; then
+              echo "${program} keyword found in the description."
+              gh issue edit "$NUMBER" --add-label "${program}"
+              echo "{has_${program}}=true"
+            else
+              echo "${program} keyword not found in the description."
+              echo "{has_${program}}=false"
+            fi
+          done
+        env:
```

Comment:
is there a need for token or other things, are we using them somewhere

Reply:
@girish-07 ^^

Reply:
No, we do not need any tokens, ensure that the appropriate labels are present @pranshugupta54